ui/src/helpers/customHooks.ts

```diff
@@ -1,19 +1,68 @@
+import axios from 'axios';
 import { Setlist, SetlistFolder } from '../types/setlist.types';
 import { SongSchema, SongViewSchema } from '../types/song.types';
-import { UserEditorProps } from '../types/user.types';
-import { useMemo } from 'react';
-import { useSelector } from 'react-redux';
+import { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { updateAxiosClient } from '../components/custom/customAxios';
+import { User } from '../types/user.types';
+import { fetchOwnership } from '../reducers';
+import { Ownership } from '../types/ownership.types';
 
 type RootState = {
-  user: UserEditorProps;
+  user: string;
   songs: SongSchema[] | SongViewSchema[];
   setlists: Setlist[];
   folders: SetlistFolder[];
+  ownership: Ownership;
 };
 
-export const useUser = () => {
-  const { token, _doc } = useSelector((state: RootState) => state.user);
-  return useMemo(() => ({ token, user: _doc }), [token, _doc]);
+export const useUser = (): { token: string; user?: User; loading: boolean } => {
+  const token = useSelector((state: RootState) => state.user);
+  const dispatch = useDispatch();
+  const [user, setUser] = useState<User>();
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        setLoading(true);
+        const { data } = await axios.post('/external-api/auth/verify-token', {
+          token: token,
+        });
+        updateAxiosClient(token);
+        setUser(data);
+        const { data: ownershipData, status } = await axios.get('/api/ownerships/get', {
+          params: {
+            userId: data.id,
+          },
+        });
+        if (status === 200 && ownershipData.length > 0) {
```

Comment:
i guess by this functionality, ownership in ROW will only be created once user first logs in to ROW. not an issue, but we have to be ready to communicate this when we launch. eg. you cant add people to setlists if they havent logged in

Reply:
yep, it will auto create if doesnt exist for first time user from our web signing in